zanata-war/src/main/java/org/zanata/action/NewProfileAction.java

```diff
@@ -92,29 +92,18 @@ public void onCreate() {
 
             //validate username if enforcing matching username is enabled
             if (applicationConfiguration.isEnforceMatchingUsernames()) {
-                if(StringUtils.isBlank(username) || isUsernameTaken(username) || !isUsernameValid(username)) {
-                    String message;
-                    String exceptionMessage;
-                    if (StringUtils.isBlank(username)) {
-                        message = msgs.get("jsf.enforce.username.isEmpty");
-                        exceptionMessage =
-                                "Server option zanata.enforce.matchingusernames is set, but username from external authentication is missing.";
-                    } else if (isUsernameTaken(username)) {
-                        message = msgs.format("jsf.enforce.username.isTaken",
-                                username);
-                        exceptionMessage =
-                                "Server option zanata.enforce.matchingusernames is set, but username from external authentication is already in use. "
-                                        + username;
-                    } else {
-                        message = msgs.format("jsf.enforce.username.isInvalid",
-                                username, USERNAME_REGEX);
-                        exceptionMessage =
-                                "Server option zanata.enforce.matchingusernames is set, but username from external authentication is not valid for Zanata. "
-                                        + username + ", valid pattern: "
-                                        + USERNAME_REGEX;
-                    }
-                    facesMessages.addGlobal(SEVERITY_ERROR, message);
-                    throw new ZanataServiceException(exceptionMessage);
+                if (StringUtils.isBlank(username)) {
+                    throw new ZanataServiceException(
+                            "Server option zanata.enforce.matchingusernames is set, but username from external authentication is missing.");
+                } else if (isUsernameTaken(username)) {
+                    throw new ZanataServiceException(
+                            "Server option zanata.enforce.matchingusernames is set, but username from external authentication is already in use."
```

Comment:
Missing a space after "use.". Also, should this be a colon, i.e. "... already in use: "?